Tasks/Common/azure-arm-rest/azure-app-service.ts

```diff
@@ -0,0 +1,185 @@
+                var publihingProfile = response.body;
+                parseString(publihingProfile, (error, result) => {
+                    for (var index in result.publishData.publishProfile) {
+                        // Currently, we need Publishing profile - MSDeploy
+                        if (result.publishData.publishProfile[index].$.publishMethod === "MSDeploy") {
+                            dataDeferred.resolve(result.publishData.publishProfile[index].$);
+                        }
+                    }
+                    dataDeferred.reject(tl.loc('ErrorNoSuchDeployingMethodExists'));
+                });
+            }
+            else {
+                dataDeferred.reject('Failed to fetch PP : ' + JSON.stringify(response));
+            }
+        }, (error) => {
+            dataDeferred.reject(error);
+        });
+
+        return dataDeferred.promise;
+    }
+
+    public async getAppSettings() {
+        var dataDeferred = Q.defer<any>();
+        var resourceGroupName: string = await this.getResourceGroupName();
+        var slotUrl: string = this.slotName ? `/slots/${this.slotName}` : '';
+        var httpRequest = new webClient.WebRequest();
+        httpRequest.method = 'POST';
+        httpRequest.uri = this.azureServiceClient.getRequestUri('//subscriptions/{subscriptionId}/resourceGroups/{ResourceGroupName}/providers/Microsoft.Web/sites/{AppServiceName}/{slotUrl}/config/appsettings/list',
+            {
+                '{ResourceGroupName}': resourceGroupName,
+                '{AppServiceName}': this.appServiceName,
+                '{slotUrl}': slotUrl
+            }, null, '2016-08-01');
+
+        this.azureServiceClient.beginRequest(httpRequest).then((response: webClient.WebResponse) => {
+            if(response.statusCode == 200) {
+                dataDeferred.resolve(response.body);
+            }
+            else {
+                dataDeferred.reject("Failed for get App Details: " + JSON.stringify(response));    
+            }
+        }, (error) => {
+            dataDeferred.reject("Failed for get App Details: " + error.toString())
+        });
+
+        return dataDeferred.promise;
+    }
+
+    public async getAppDetails(force?: boolean) {
+        var dataDeferred = Q.defer<any>();
+        if(this.appDetails && !force) {
+            console.log("getAppDetails in cache");
+            dataDeferred.resolve(this.appDetails);
+        }
+        else {
+            console.log(" getAppDetails not in cache");
+            var resourceGroupName: string = await this.getResourceGroupName();
+            var slotUrl: string = this.slotName ? `/slots/${this.slotName}` : '';
+            var httpRequest = new webClient.WebRequest();
+            httpRequest.method = 'GET';
+            httpRequest.uri = this.azureServiceClient.getRequestUri('//subscriptions/{subscriptionId}/resourceGroups/{ResourceGroupName}/providers/Microsoft.Web/sites/{AppServiceName}/{slotUrl}',
+                {
```

Comment:
i think using `` will serve the same purpose as using getRequestUri.

Reply:
This function also removes additional slashes and update the URI with subscription ID.
and also encodes the URI